### PR TITLE
<main> Improving workloads tests

### PR DESCRIPTION
**Pull request by lscalabrini01**
Backports:
v2.9 https://github.com/rancher/rancher/pull/47477
v2.8 https://github.com/rancher/rancher/pull/47480
v2.7 https://github.com/rancher/rancher/pull/47482

### CODE REVIEW

**Review by slickwarren — COMMENTED** [review]
these changes at a high level look fine but there were some jenkins failures that will have to be investigated before approval

**Review by floatingman — DISMISSED** [review]
This looks good.

File: tests/v2/validation/workloads/deployment_test.go
Comment (floatingman) [inline tests/v2/validation/workloads/deployment_test.go:80]: One thing I'm noticing is that this function is called repeatedly with the same arguments. I wonder if it can be less repeated by creating another function with less parameters?

File: tests/v2/validation/workloads/deployment_test.go
Comment (floatingman) [inline tests/v2/validation/workloads/deployment_test.go:72]:
```suggestion
	func newRedisContainerTemplate(containerName string) *corev1.Container {
    return workloads.NewContainer(containerName,
        redisImageName,
        corev1.PullAlways,
        []corev1.VolumeMount{},
        []corev1.EnvFromSource{},
        nil,
        nil,
        nil,
    )
}
```

File: tests/v2/validation/workloads/deployment_test.go
Comment (lscalabrini01) [reply tests/v2/validation/workloads/deployment_test.go:80]: Yes, I can

Comment (floatingman) [reply tests/v2/validation/workloads/deployment_test.go:80]: Maybe like this?
```
func newRedisContainerTemplate(containerName string) *corev1.Container {
    return workloads.NewContainer(containerName,
        redisImageName,
        corev1.PullAlways,
        []corev1.VolumeMount{},
        []corev1.EnvFromSource{},
        nil,
        nil,
        nil,
    )
}
```

Comment (lscalabrini01) [reply tests/v2/validation/workloads/deployment_test.go:80]: Maybe it could take the image name as a parameter

Comment (anupama2501) [reply tests/v2/validation/workloads/deployment_test.go:80]: Based on our previous design discussion, we want to avoid creating multiple layers of abstraction. Since workloads.NewContainer is already an abstraction, creating another one could cause confusion. Instead, we can repeat the same steps directly in the tests to improve readability.

Comment (lscalabrini01) [reply tests/v2/validation/workloads/deployment_test.go:80]: So, could we keep it the same way? @floatingman @anupama2501

Comment (anupama2501) [reply tests/v2/validation/workloads/deployment_test.go:80]: yes IMHO.

Comment (floatingman) [reply tests/v2/validation/workloads/deployment_test.go:80]: It is a maintenance hardship since the parameters are the same across the three functions, but it's really not that important.

**Review by floatingman — DISMISSED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]

File: tests/v2/actions/workloads/pods/pods.go
Comment (slickwarren) [inline tests/v2/actions/workloads/pods/pods.go:138]: based on our discussion I would reduce this time to 1

Comment (lscalabrini01) [reply tests/v2/actions/workloads/pods/pods.go]: done

Comment (lscalabrini01) [reply tests/v2/actions/workloads/pods/pods.go]: @slickwarren I've tested and 1 second is few time, I am getting timeout error now, could be 5 seconds? it working fine

**Review by slickwarren — APPROVED** [review]

**Review by floatingman — APPROVED** [review]
LGTM

**Pull request merged**
